### PR TITLE
tmt: Add tmt test export to shell format.

### PR DESCRIPTION
**Pull request by RHEmployee**
Provide data dict in form which is better formated by shell.
Example of format:

key: value
name: test1
\n
\n
key: value
name: test2

If more tests are present \n\n sequence is used as a delimiter.

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -735,6 +736,24 @@ def context_to_dict(context):
...
+    # List available attributes
+    for key, value in sorted(data.items()):
+        output += "{0}: ".format(fmf.utils.color(key, 'green'))
+        if isinstance(value, type("")):
+            output += value.rstrip("\n")
+        elif isinstance(value, list) and all(
+                [isinstance(item, type("")) for item in value]):
+            output += fmf.utils.listed(value)
+        else:
+            output += pretty(value)
```
Comment (lukaszachy) [inline tmt/utils.py:752]: We should find a nice representation for a dict as it can be present even in require/recommend (hint: library)

File: tmt/utils.py [hunk 2]
```diff
@@ -735,6 +736,24 @@ def context_to_dict(context):
         for key, value in environment_to_dict(context).items()}
 
 
+def dict_to_shell(data, width=None, sort=False):
+    if not data:
+        return "\n"
+    output = ""
+    # List available attributes
+    for key, value in sorted(data.items()):
+        output += "{0}: ".format(fmf.utils.color(key, 'green'))
```
Comment (lukaszachy) [inline tmt/utils.py:745]: Do we need colored output for shell?

File: tmt/utils.py (hunk 1 above)
Comment (lukaszachy) [reply tmt/utils.py:752]: Example of a test using require with dict (library)
```
recommend: mariadb and mariadb-server
require: [{'name': '/mariadb',
  'ref': 'master',
  'url': 'https://github.com/beakerlib/database'},
 'expect',
 'beakerlib']
result: respect
```

**Comment by lukaszachy** [comment]
Also we need new test for this format - ideally exporting complicated metadata (dict, quoting...)

**Review by lukaszachy — CHANGES_REQUESTED** [review]
What is the expected usage? I'm not sure about the resulting format as we have dictionaries there.

File: tmt/utils.py (hunk 2 above)
Comment (RHEmployee) [reply tmt/utils.py:745]: Dunno. This is basically stolen from fmf codebase. But I can remove it.

**Comment by RHEmployee** [comment]
> What is the expected usage? I'm not sure about the resulting format as we have dictionaries there.

To have freedom to work with data in another form? However, it looks like yq and yaml output can be way how to reach goals also.

**Comment by lukaszachy** [comment]
> > What is the expected usage? I'm not sure about the resulting format as we have dictionaries there.
> 
> To have freedom to work with data in another form? However, it looks like yq and yaml output can be way how to reach goals also.

Let me rephrase - How should dictionary be presented?

**Comment by RHEmployee** [comment]
Hi. Work on reopening of this MR. My fault. Close of this MR was not intended. Just force pushing around to my devel master branch on different repo :(. Ok. So now I know how github behaves in this situation.

**Comment by RHEmployee** [comment]
Reopening again. Hope there will be way how to change branch in opened state.

**Comment by psss** [comment]
What are the plans with this pull request? Is the use case still valid?

**Comment by RHEmployee** [comment]
Hello. Oh. This commit again. Sorry I made cleanup of my tmt fork and I just forgot about this one.
We should leave (or switch) it in closed state.

**Comment by psss** [comment]
Ok, closing.
